parquet/src/main/java/org/apache/iceberg/parquet/Parquet.java

```diff
@@ -231,6 +241,46 @@ private WriteBuilder createContextFunc(
       return this;
     }
 
+    private void setBloomFilterConfig(
+        Context context, BiConsumer<String, Boolean> withBloomFilterEnabled) {
+      Splitter dotSplitter = Splitter.on('.');
+      Joiner dotJoiner = Joiner.on(".");
+      context
+          .columnBloomFilterEnabled()
+          .forEach(
+              (columnPath, value) -> {
+                // check the column exists and support bloom filter
+                Types.NestedField field = schema.caseInsensitiveFindField(columnPath);
+                if (field == null) {
+                  LOG.warn(
+                      "Invalid bloom filter column configuration, field: {} is not found in the schema: {}, "
+                          + "this column configuration will be ignored.",
+                      columnPath,
+                      schema);
+                  return;
+                }
+                Type fieldType = field.type();
+                // doesn't support boolean, list, map, struct type
+                if (!fieldType.isPrimitiveType()
+                    || fieldType.typeId().equals(Type.TypeID.BOOLEAN)) {
+                  LOG.warn(
+                      "Invalid bloom filter column configuration: field {} is of type '{}' which cannot "
+                          + "benefit from a bloom filter. This column configuration will be ignored.",
+                      columnPath,
+                      fieldType);
+                  return;
+                }
+
+                String compatibleName =
+                    dotJoiner.join(
+                        Streams.stream(dotSplitter.split(columnPath))
+                            .map(AvroSchemaUtil::makeCompatibleName)
+                            .collect(Collectors.toList()));
```

Comment:
This should be separated into a util method in `ParquetSchemaUtil` that can be tested.